src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -289,13 +290,16 @@ private void MethodChecks(MethodDeclarationSyntax syntax, Binder withTypeParamsB
 
             if (IsPartial)
             {
-                // check that there are no out parameters in a partial
-                foreach (var p in this.Parameters)
+                if (MessageID.IDS_FeatureExtendedPartialMethods.GetFeatureAvailabilityDiagnosticInfo(DeclaringCompilation) is { } info)
```

Comment:
>if (MessageID.IDS_FeatureExtendedPartialMethods.GetFeatureAvailabilityDiagnosticInfo(DeclaringCompilation) is { } info) [](start = 16, length = 119)

Consider moving this inside `if (p.RefKind == RefKind.Out) { ... }`.

Reply:
Moved the check to 'CheckModifiers'.